### PR TITLE
Add Value.Format

### PR DESCRIPTION
**Pull request by dsnet**
This formats a HuJSON AST.

### CODE REVIEW

**Comment by dsnet** [comment]
The non-test logic is in a good state. The tests need a lot more coverage.

**Comment by crawshaw** [comment]
I ran all of our ACLs through this formatter. Some ntoes:
- Every single ACL changes, because the whitespace used to indent has changed.
- If I remove all the whitespace, newlines, and commas, every ACL file is unchanged. That's good news, no bugs! This code is not even mangling any comments.
- Experimenting with different amounts of whitespace removal and looking at samples, it seems reasonably common to change the number of newlines in an ACL file. Somewhere from a quarter to half of all ACLs change.
- The formatter seems somewhat aggressive about adding trailing commas. It does not seem to always do it! E.g. if I add this test:
  ```
  in:   "{\"name\": \"value\"\n}",
  want: "{\n\t\"name\": \"value\"\n}",
  ```
  the formatter reformats this without adding a trailing comma. However this test case adds two trailing commas:
  ```
        in: `{
        "Tests": [
                {
                        "User": "a",
                        "Allow": ["a"]
                },
                // {
                //  "User": "b",
                //  "Allow": ["b"]
                // },
                // {
                //  "User": "c",
                //  "Allow": ["c"]
                // }
        ]
  }`,
        want: `{
        "Tests": [
                {
                        "User":  "a",
                        "Allow": ["a"],
                },
                // {
                //  "User": "b",
                //  "Allow": ["b"]
                // },
                // {
                //  "User": "c",
                //  "Allow": ["c"]
                // }
        ],
  }`,
  ```
  It's not clear to me what the right thing to do here is. This is possibly fine?

File: format.go [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+
+		// Update the block statistics.
+		lineLengths = append(lineLengths, lineLength)
+		stats = lineStats{
+			firstLength: lineLengths[0],
+			lastLength:  lineLengths[len(lineLengths)-1],
+			multiline:   len(lineLengths) > 1,
+		}
+		for i := 0; !expand && i < len(lineLengths); i++ {
+			expand = lineLengths[i] > 80
```
Comment (crawshaw) [inline format.go:165]: `const punchCardWidth = 80`

**Review by crawshaw — APPROVED** [review]

**Comment by dsnet** [comment]
> The formatter seems somewhat aggressive about adding trailing commas. It does not seem to always do it!

I believe this is because of the interaction of two heuristics:
1. If the input is already standard JSON, then the format guarantees that the output will continue to be standard JSON. Since standard JSON does not have trailing commas, the output will never have trailing commas.
2. If the input is HuJSON, then we always put trailing commas for JSON objects and arrays that span multiple lines (but never for JSON objects/arrays that are on a single line.

I'm open to any adjustments of the rules.

**Comment by dsnet** [comment]
I'll finish polishing up the unit tests and submit this tomorrow.

**Pull request merged**